cli/src/ui.rs

```diff
@@ -632,14 +621,54 @@ impl Ui {
             Some(false) => "(yN)",
             None => "(yn)",
         };
-        let default_index = default.map(|c| if c { 0 } else { 1 });
-
-        let index = self.prompt_choice(
+        self.prompt_choice_with(
             &format!("{prompt} {default_str}"),
-            &["y", "n", "yes", "no", "Yes", "No", "YES", "NO"],
-            default_index,
-        )?;
-        Ok(index % 2 == 0)
+            default.map(|v| if v { "y" } else { "n" }),
+            |input| {
+                if input.eq_ignore_ascii_case("y") || input.eq_ignore_ascii_case("yes") {
+                    Ok(true)
+                } else if input.eq_ignore_ascii_case("n") || input.eq_ignore_ascii_case("no") {
+                    Ok(false)
+                } else {
+                    Err("unrecognized response")
+                }
+            },
+        )
+    }
+
+    /// Repeats the given prompt until `parse(input)` returns a value.
+    ///
+    /// If the default `text` is given, an empty input will be mapped to it. It
+    /// will also be used in non-interactive session. The default `text` must
+    /// be parsable. If no default is given, this function will fail in
+    /// non-interactive session.
+    pub fn prompt_choice_with<T, E: fmt::Debug + fmt::Display>(
+        &self,
+        prompt: &str,
+        default: Option<&str>,
+        mut parse: impl FnMut(&str) -> Result<T, E>,
+    ) -> io::Result<T> {
+        if !Self::can_prompt() {
```

Comment:
(Optional) Perhaps add a `debug_assert` that the default is parseable, to catch this error earlier?

Reply:
made parsing mandatory. we don't care the parsing cost here.